### PR TITLE
Add: clarify config relationship; fix headings.

### PR DESCRIPTION
**Pull request by teesloane**
Clarify the relationship between "step vs. job vs. workflow vs. pipeline".  Research has found that these concepts are a little unclear to users. 

In this PR:
- Move the "configuration" section up in the document.
- Some re-wording in the section under `#configuration` (although @rosieyohannan has done lots of great work on this document recently as well, especially the graphics). 
- Cleans up the table of contents to not be nested. 

@KateCatlin is wondering if we can get some other designs that can better represent the relationship between steps/jobs/workflow/pipelines, but for now, hopefully this prose clears it up.

### CODE REVIEW

**Comment by rosieyohannan** [comment]
Looks great! 🎉

File: jekyll/_cci2/concepts.md [hunk 1]
```diff
@@ -21,14 +21,34 @@ On the Projects Dashboard, you can either:
...
 
-{:.tab.addprojectpage.Cloud}
 ![header]({{ site.baseurl }}/assets/img/docs/CircleCI-2.0-setup-project-circle101_cloud.png)
 
-{:.tab.addprojectpage.Server}
-![header]({{ site.baseurl }}/assets/img/docs/CircleCI-2.0-setup-project-circle101.png)
+## Configuration
 
-## User Types
+CircleCI believes in *configuration as code*. Your entire continuous integration/deployment process is orchestrated through a single file called `config.yml`.  The `config.yml` file is located in a folder called `.circleci` at the root of your project. CircleCI uses the YAML syntax for config, see the [Writing YAML]({{ site.baseurl }}/2.0/writing-yaml/) document for basics.
```
Comment (michelle-luna) [inline jekyll/_cci2/concepts.md:28]:
```suggestion
CircleCI believes in *configuration as code*. Your entire continuous integration and deployment process is orchestrated through a single file called `config.yml`.  The `config.yml` file is located in a folder called `.circleci` at the root of your project. CircleCI uses the YAML syntax for config, see the [Writing YAML]({{ site.baseurl }}/2.0/writing-yaml/) document for basics.
```

File: jekyll/_cci2/concepts.md [hunk 2]
```diff
@@ -21,14 +21,34 @@ On the Projects Dashboard, you can either:
...
+
+```bash
+├── .circleci
+│   ├── config.yml
+├── README
+└── all-your-projects-other-files-and-folders
```
Comment (michelle-luna) [inline jekyll/_cci2/concepts.md:34]:
```suggestion
└── all-other-project-files-and-folders
```

File: jekyll/_cci2/concepts.md [hunk 3]
```diff
@@ -21,14 +21,34 @@ On the Projects Dashboard, you can either:
...
+```
 
+`config.yml` is a powerful YAML file that defines the entire pipeline for your project. For a full overview of the various keys that can be used see the [Configuration Reference]({{ site.baseurl }}/2.0/configuration-reference/).
```
Comment (michelle-luna) [inline jekyll/_cci2/concepts.md:37]:
```suggestion
`config.yml` is a powerful YAML file that defines the entire pipeline for your project. For a full overview of the various keys that are used, see the [Configuration Reference]({{ site.baseurl }}/2.0/configuration-reference/). 
```

File: jekyll/_cci2/concepts.md [hunk 4]
```diff
@@ -21,14 +21,34 @@ On the Projects Dashboard, you can either:
...
+```bash
+├── .circleci
+│   ├── config.yml
+├── README
+└── all-your-projects-other-files-and-folders
+```
 
+`config.yml` is a powerful YAML file that defines the entire pipeline for your project. For a full overview of the various keys that can be used see the [Configuration Reference]({{ site.baseurl }}/2.0/configuration-reference/). 
+
+While your CircleCI configuration can be adapted to fit many different needs, the following terms, sorted in order of granularity and dependence, describe the anatomy of most common CircleCI projects:
```
Comment (michelle-luna) [inline jekyll/_cci2/concepts.md:39]:
```suggestion
Your CircleCI configuration can be adapted to fit many different needs of your project. The following terms, sorted in order of granularity and dependence, describe the components of most common CircleCI projects:
```

**Comment by teesloane** [comment]
Thanks @michelle-luna !

**Pull request merged**